src/Microsoft.AspNetCore.Routing/Builder/MapEndpointEndpointDataSourceBuilderExtensions.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matching;
+using Microsoft.AspNetCore.Routing.Patterns;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    public static class MapEndpointEndpointDataSourceBuilderExtensions
+    {
+        public static EndpointBuilder MapEndpoint(
+            this EndpointDataSourceBuilder builder,
+            Func<RequestDelegate, RequestDelegate> invoker,
+            string name,
+            string pattern)
+        {
+            return MapEndpoint(builder, invoker, name, pattern, null);
+        }
+
+        public static EndpointBuilder MapEndpoint(
+            this EndpointDataSourceBuilder builder,
+            Func<RequestDelegate, RequestDelegate> invoker,
+            string name,
+            RoutePattern pattern)
+        {
+            return MapEndpoint(builder, invoker, name, pattern, null);
+        }
+
+        public static EndpointBuilder MapEndpoint(
+            this EndpointDataSourceBuilder builder,
+            Func<RequestDelegate, RequestDelegate> invoker,
+            string name,
+            string template,
+            IList<object> metadata)
```

Comment:
`params object[]` ?

Reply:
Yeah could do. I worry a little about disambiguating overloads that involve `params object[]` though.